apps/desktop-wallet/src/api/apiDataHooks/useToken.ts

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { NFT } from '@alephium/shared'
+import { explorer } from '@alephium/web3'
+import { useQuery } from '@tanstack/react-query'
+import { useMemo } from 'react'
+
+import useFTList from '@/api/apiDataHooks/useFTList'
+import { fungibleTokenMetadataQuery, nftDataQuery, nftMetadataQuery, tokenTypeQuery } from '@/api/queries/tokenQueries'
+import { ListedFT, NonStandardToken, TokenId, UnlistedFT } from '@/types/tokens'
+
+type Token = {
+  data: ListedFT | UnlistedFT | NFT | NonStandardToken
+  isLoading: boolean
+}
+
+const useToken = (id: TokenId): Token => {
```

Comment:
So, I think you will be aligned with what's happening in this hook.
1. We get the fungible token list. If our token is there, all following queries will be skipped and we'll return a structurally shared object
2. If the token wasn't there (and only then, thanks to `skip`) we query for its type. This will determine which queries will be executed in the next component re-render.
3. In the next re-render, since the token is not a listed one, we have its type, so we query for its metadata.
4. Finally we fetch for the nft data if the type is NFT and we have received the metadata.

Some notes:
1. Any component that uses this hook will render at least 2 times on app launch
   1. 1st time the `isLoadingFtList` will be `true` and the hook will not run any other queries
   2. 2nd time the token list will be available so if the token is part of it, no more re-renders.
2. If the id is of an NFT, the component that uses this hook will be re-render at least 5 times.
```
   1. fTList: undefined,         isLoadingFtList: true
   2. fTList: [...],             isLoadingFtList: false,     tokenType: undefined,   isLoadingTokenType: true
   3. tokenType: 'non-fungible', isLoadingTokenType: false,  nftMetadata: undefined, isLoadingNftMetadata: true
   4. nftMetadata: {...},        isLoadingNftMetadata: false, nftData: undefined,    isLoadingNftData: true
   5. nftData: {...},            isLoadingNftData: false
```